app/components/viewer/TimeLine.tsx

```diff
@@ -0,0 +1,175 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable import/prefer-default-export */
+import { useEffect, useRef, useState } from 'react';
+import * as THREE from 'three';
+import {
+    Billboard, Sphere, Text, Plane,
+} from '@react-three/drei';
+import { useFrame } from '@react-three/fiber';
+import {
+    Slider, InputNumber, Row, Col, Button,
+} from 'antd';
+import { ReplayData } from '../../lib/api/apiRequests';
+import { getRaceTimeStr, timeDifference } from '../../lib/utils/time';
+
+export class TimeLineInfos {
+    currentRaceTime: number;
+    followedReplay: any;
+    constructor() {
+        this.currentRaceTime = 0;
+        this.followedReplay = null;
+    }
+}
+
+interface TimeLineViewProps {
+    replaysData: ReplayData[];
+    timeLineGlobal: TimeLineInfos;
+}
+
+let playInterval: any;
+let playing = false;
+let prevReplayCount: number = 0;
+let prevFollowedReplay: any = null;
```

Comment:
I would've made these state variables - what's the reason for making them variables outside of the component?